src/map.cpp

```diff
@@ -6506,6 +6583,19 @@ void map::loadn( const int gridx, const int gridy, const int gridz, const bool u
             return;
         }
     }
+    // Adding snow to new submaps
+    for( int j = 0; j < SEEY; j++ ) {
+        for( int i = 0; i < SEEX; i++ ) {
+            const tripoint pnt( gridx * SEEX + i, gridy * SEEY + j, gridz );
+            const point p( i, j );
+            if( ( !tmpsub->get_ter( p ).obj().has_flag( TFLAG_INDOORS ) ) &&
+                ( !tmpsub->get_ter( p ).obj().has_flag( "TREE" ) )  &&
+                ( tmpsub->get_furn( p ) == f_null ) && ( is_outside( pnt ) ) &&
+                x_in_y( percent_settled, 100 ) ) {
```

Comment:
1. Excessive parenthesis.
2. `has_flag` are presumably the most expensive checks here, they should go last.
3. `obj()` can be omitted, simply `get_ter( p )->has_flag( TFLAG_INDOORS )` would work too.
4. Not quite sure about the `"TREE"` flag (looks like a hack which doesn't cover all possible cases).